src/crashhandler/main.cpp

```diff
@@ -0,0 +1,99 @@
+/***************************************************************************
+                            crssync.cpp
+                            -------------------
+   begin                : October 2017
+   copyright            : (C) 2017 by Nathan Woodrow
+   email                : woodrow.nathan@gmail.com
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include <iostream>
+
+#define _NO_CVCONST_H
+#define _CRT_STDIO_ISO_WIDE_SPECIFIERS
+
+#include <QApplication>
+#include <QMainWindow>
+#include "qgscrashdialog.h"
+#include "qgsstacktrace.h"
+#include "qgscrashreport.h"
+
+
+int main( int argc, char *argv[] )
+{
+  if ( argc < 2 )
+  {
+    std::cout << "QGIS Crash Handler Usage: \n"
+              << "qgscrashhandler {infofile}" << std::endl;
+    return -1;
+  }
+
+  QApplication app( argc, argv );
+  app.setQuitOnLastWindowClosed( true );
+  QCoreApplication::setOrganizationName( "QGIS" );
+  QCoreApplication::setApplicationName( "QGIS3" );
+
+  QString extraInfoFile = QString( argv[1] );
+  std::cout << "Extra Info File: " << extraInfoFile.toUtf8().data() << std::endl;
+
+  QFile file( extraInfoFile );
+  QString processIdString;
+  QString threadIdString;
+  QString exceptionPointersString;
+  QString symbolPaths;
+  QString reloadArgs;
+  QStringList versionInfo;
+
+  if ( file.open( QIODevice::ReadOnly | QIODevice::Text ) )
+  {
+    processIdString = file.readLine();
+    threadIdString = file.readLine();
+    exceptionPointersString = file.readLine();
+    symbolPaths = file.readLine();
+    reloadArgs = file.readLine();
+    // The version info is the last stuff to be in the file until the end
+    // bit gross but :)
+    QString info = file.readAll();
+    versionInfo = info.split("\n");
+  }
```

Comment:
Graceful exit if we can't open the file?

Reply:
I might still show the dialog but with no info.

Reply:
although this shouldn't really happen anyway but dialog might be better then just crashing out.

Reply:
With no info this is what you get:

![image](https://user-images.githubusercontent.com/381660/32422596-e5e0de80-c2ec-11e7-85d7-c3829cf429d1.png)

I think that is ok to leave like that.

Reply:
Looks perfect! Just needs a . between "information Your"

Also "Stack trace unable to be generated" should be "Stack trace could not be generated"